go/simple/v2/ampli/ampli.go

```diff
@@ -0,0 +1,338 @@
+		EventType: "Event Object Types",
+		EventProperties: map[string]interface{}{
+			"requiredObject":      stronglyTypedEvent.RequiredObject,
+			"requiredObjectArray": stronglyTypedEvent.RequiredObjectArray,
+		},
+	}
+}
+
+type Ampli struct {
+	Disabled bool
+	Client   amplitude.Client
+}
+
+// Load initializes the Ampli wrapper.
+// Call once when your application starts.
+func (a *Ampli) Load(options LoadOptions) {
+	a.Disabled = options.Disabled
+
+	if a.Client != nil {
+		log.Default().Printf("Warn: Ampli is already initialized. Ampli.load() should be called once at application start up.")
+
+		return
+	}
+
+	if options.Client.isEmpty() {
+		options.Client = LoadClientOptions{
+			Configuration: DefaultConfiguration,
+		}
+	}
+
+	var apiKey string
+	if options.Client.APIKey != "" {
+		apiKey = options.Client.APIKey
+	} else if options.Environment != "" {
+		apiKey = APIKey[options.Environment]
+	} else if options.Client.Configuration.APIKey != "" {
+		apiKey = options.Client.Configuration.APIKey
+	}
+
+	if !(apiKey != "" || options.Client.Instance != nil) {
+		log.Default().Printf("Error: Ampli.Load() requires option.Environment, " +
+			"and apiKey from either options.Client.APIKey or APIKey[options.Environment], " +
+			"or options.Client.Instance")
+	}
+
+	var configuration amplitude.Config
+	if !options.Client.Configuration.IsEmpty() {
+		configuration = options.Client.Configuration
+	} else {
+		configuration = DefaultConfiguration
+	}
+
+	if configuration.Plan == (amplitude.Plan{}) {
+		configuration.Plan = DefaultConfiguration.Plan
+	}
+
+	if options.Client.Instance != nil {
+		a.Client = options.Client.Instance
+	} else {
+		configuration.APIKey = apiKey
+		a.Client = amplitude.NewClient(configuration)
+	}
+}
+
+// InitializedAndEnabled checks if Ampli is initialized and enabled.
+func (a *Ampli) InitializedAndEnabled() bool {
+	return !a.Disabled
+}
+
+// Track tracks an StronglyTypedEvent.
+func (a *Ampli) Track(userID string, event StronglyTypedEvent, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	if userID != "" {
+		eventOptions.UserID = userID
+	}
+
+	baseEvent := event.ToEvent()
+	baseEvent.EventOptions = eventOptions
+
+	a.Client.Track(baseEvent)
+}
+
+// Identify identifies a user and set user properties.
+func (a *Ampli) Identify(userID string, identify Identify, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	if userID != "" {
+		eventOptions.UserID = userID
+	}
+
+	identifyEvent := amplitude.Event{
+		EventType: amplitude.IdentifyEventType,
+		EventProperties: map[string]interface{}{
+			"requiredNumber": identify.RequiredNumber,
+			"optionalArray":  identify.OptionalArray,
+		},
+		EventOptions: eventOptions,
+	}
+
+	a.Client.Track(identifyEvent)
+}
+
+// GroupIdentify identifies a group and set group properties.
+func (a *Ampli) GroupIdentify(groupType string, groupName string, group Group, eventOptions amplitude.EventOptions) {
```

Comment:
GroupIdentify also requires a `userId`

Reply:
It seems other SDKs don't accept `userId`. And `userId` can be passed in in `EventOptions`
- Python: https://github.com/amplitude/ampli-examples/blob/4d58124347800d4bcaa4f1045ad2e12fcbb3ec36/python/simple/v1/ampli/ampli.py#L549
- Kotlin: https://github.com/amplitude/ampli-examples/blob/4d58124347800d4bcaa4f1045ad2e12fcbb3ec36/android/kotlin/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt#L583
- TypeScript: https://github.com/amplitude/Amplitude-TypeScript/blob/c701a2b7ada6c8817bf28894f4d68cfa5dd0c403/packages/analytics-core/src/core-client.ts#L60